src/pages/index.js

```diff
@@ -100,36 +148,115 @@ function fillProfileForm() {
     descInput.value = info.description;
 }
 
-function updateInfo(event, data) {
-    userInfo.setUserInfo({
+async function updateInfo(event, data) {
+    const newInfo = {
         name: data["name-input"],
         description: data["description-input"],
+    };
+    api.editProfile({
```

Comment:
* When I edit the profile data the button text doesn't change to `Saving...`


To make it work you need to **return** the promise here. So, just add `return`


```js
    return api.editProfile({


```